extensions/scarb-doc/src/main.rs

```diff
@@ -12,6 +12,13 @@ use scarb_doc::generate_language_elements_tree_for_package;
 struct Args {
     #[command(flatten)]
     packages_filter: PackagesFilter,
+
+    /// Print information collected from packages to stdout in JSON format instead of generating a
+    /// documentation.
+    /// This feature may be useful if you want to generate documentation files by yourself.
+    /// The precise output structure is not guaranteed to be stable.
+    #[arg(long)]
+    unstable_json_output: bool,
```

Comment:
I suggest using an enum here under a flag `--output-format` so that you'll have room to add `html` output etc.

Reply:
Actually good idea, I guess I will just document this enum variant as unstable. I will dump it to a file to be consistent with other formats (md, html)